app/models/manageiq/providers/autosde/storage_manager/autosde_client.rb

```diff
@@ -70,8 +70,11 @@ def login
     data = self.AuthenticationApi.token_auth_post(auth_request, opts)
     @token = data.token
   rescue AutosdeOpenapiClient::ApiError => e
-    error_data = JSON.parse(e.response_body)
-    raise AUTH_ERRR_MSG + error_data.dig("detail", "non_field_errors")&.first
+    error_data = e.to_s
+    if e.response_body
+      error_data = JSON.parse(e.response_body).dig("detail", "non_field_errors")&.first
+    end
```

Comment:
Hm if you're going to overwrite `error_data` maybe better to do something like:
```suggestion

    error_data = if e.response_body
      JSON.parse(e.response_body).dig("detail", "non_field_errors")&.first
    else
      e.to_s
    end
```